### ISSUE TITLE
Data not automatically downloaded into new "inputadata"

### ISSUE THREAD

**Issue by mrnorman** [ESMCI/cime#1282]
1. When I try to run ./case.setup, it complains that some of the .nc files are not there.
2. If I run ./check_input_data --download before running ./case.setup, it doesn't do anything.
3. I end up having to run "./case.setup; ./check_input_data --download" multiple times to get ./case.setup to succeed

Some more specific documentation is at the link below:

https://acme-climate.atlassian.net/wiki/display/PERF/Running+ACME+on+Ubuntu+Laptop

**Comment by jgfouca** [comment]
@mrnorman one thing that would help me is if you gave me the exact case you tried.

**Comment by mrnorman** [comment]
./create_newcase -compset FC5AV1C-L -res ne4_ne4 -case ne4_v1 -mach scrollbuttons -pecount 4 -q myqueue -compiler gnu

**Comment by rljacob** [comment]
A possibly important point in this:  Matt is installing ACME on his laptop and has a completely empty input directory.

**Comment by mvertens** [comment]
When you say "multiple times" - do you need to run check_input_data
separately for *each* missing file?

**Comment by jgfouca** [comment]
OK, I will debug this by changing my input data location to an empty directory.

**Comment by mvertens** [comment]
The behavior we want is to have check_input_data load all missing files at
once onto local disk - not just one at a time.

**Comment by jgfouca** [comment]
@mvertens I strongly agree.

**Comment by mrnorman** [comment]
It wasn't one at a time, but it wasn't all at one time either. It looked to me like running ./case.setup somehow "registered" some files for download, then failed by complaining they aren't there. Then ./check_input_data --download downloads those file. Then I run ./case.setup again, and it seems to register more files for download and failing again because they aren't there. So I run ./check_input_data --download again to download those. Then one more cycle, and for my particular case, that was enough to get ./case.setup to succeed.

And yes, my inputdata directory was initially empty.

**Comment by jgfouca** [comment]
@mrnorman I took a look at this and addressed it as best I could from the CIME side. The real problem here is that the buildnml scripts for clm make assumptions about the state of the input area without calling check_input_data ( @singhbalwinder ) . The CIME infrastructure itself makes no promises about input data until the RUN phase (not the SETUP phase). The PR I made to address this ticket allows users to easily change the input-root via create_test, so it will at least be easy to test this kind of thing in the future.
